badge-maker/lib/badge-renderers.js

```diff
@@ -678,24 +726,52 @@ function forTheBadge({
     }
   }
 
+  const renderedBackground = needsLabelRect
+    ? `
+        <rect
+          width="${labelRectWidth}"
+          height="${BADGE_HEIGHT}"
+          fill="${escapeXml(outLabelColor)}"
+        />
+        <rect
+          x="${labelRectWidth}"
+          width="${messageRectWidth}"
+          height="${BADGE_HEIGHT}"
+          fill="${escapeXml(color)}"
+        />
+      `
+    : `
+        <rect
+          width="${messageRectWidth}"
+          height="${BADGE_HEIGHT}"
+          fill="${escapeXml(color)}"
+        />
```

Comment:
I found this ternary/template spread over ~20 lines a bit hard to parse. How about we wrap the templates so we do something like

```js
const labelRect = `<rect...`
const messageRect = `<rect...`
const renderedBackground = needsLabelRect ? labelRect + messageRect : messageRect
```

or

```js
const renderLabelRect = function(...
const renderMessageRect = function(...
const renderedBackground = needsLabelRect ? renderLabelRect() + renderMessageRect() : renderMessageRect()
```

to keep the actual ternary short?

Reply:
The message rect markup is a bit different for the two cases.

`<rect width="${messageRectWidth}"`

vs

`<rect x="${labelRectWidth}" width="${messageRectWidth}"`

What about replacing the many-line ternary with an `if`/`else`?

```js
let renderedBackground
if (needsLabelRect) {
  renderedBackground = ...
} else {
  renderedBackground = ...
}
```

Reply:
Yeah sounds good. That would make it clearer :+1: